@here/harp-omv-datasource/lib/OmvDecodedTileEmitter.ts

```diff
@@ -380,30 +392,111 @@ export class OmvDecodedTileEmitter implements IOmvEmitter {
+            // caps.
+            if (hasIndividualLineSegments) {
+                localLineSegments = [];
+
+                // Compute length of whole line and offsets of individual segments.
+                let lineLength = 0;
+                if (polyline.positions.length > 1) {
+                    const pointCount = polyline.positions.length;
+
+                    let lastSegmentOffset = 0;
+
+                    for (let i = 0; i < pointCount - 1; i++) {
+                        const localLine: number[] = [];
+                        const worldLine: number[] = [];
+                        const lineUvs: number[] = [];
+                        const segmentOffsets: number[] = [];
+
+                        const pos1 = polyline.positions[i];
+                        const pos2 = polyline.positions[i + 1];
+                        webMercatorTile2TargetWorld(extents, this.m_decodeInfo, pos1, tmpV3);
+                        worldLine.push(tmpV3.x, tmpV3.y, tmpV3.z);
+                        webMercatorTile2TargetWorld(extents, this.m_decodeInfo, pos2, tmpV4);
+                        worldLine.push(tmpV4.x, tmpV4.y, tmpV4.z);
+
+                        if (computeTexCoords) {
+                            {
+                                const { u, v } = computeTexCoords(pos1, extents);
+                                lineUvs.push(u, v);
+                            }
+                            {
+                                const { u, v } = computeTexCoords(pos2, extents);
+                                lineUvs.push(u, v);
+                            }
+                        }
+                        if (hasUntiledLines) {
+                            // Find where in the [0...1] range relative to the line our current
+                            // vertex lies.
+                            let offset =
+                                this.findRelativePositionInLine(tmpV3, untiledLine) / lineDist;
+                            segmentOffsets.push(offset);
+                            offset = this.findRelativePositionInLine(tmpV4, untiledLine) / lineDist;
+                            segmentOffsets.push(offset);
+                        } else {
+                            segmentOffsets.push(lastSegmentOffset);
+
+                            // Compute length of segment and whole line to scale down later.
+                            const segmentLength = tmpV3.distanceTo(tmpV4);
+                            lineLength += segmentLength;
+                            lastSegmentOffset += segmentLength;
+                            segmentOffsets.push(lastSegmentOffset);
+                        }
+
+                        tmpV3.sub(this.m_decodeInfo.center);
+                        localLine.push(tmpV3.x, tmpV3.y, tmpV3.z);
+                        tmpV4.sub(this.m_decodeInfo.center);
+                        localLine.push(tmpV4.x, tmpV4.y, tmpV4.z);
+
+                        localLineSegments!.push(localLine);
```

Comment:
Maybe I see something wrong (because of web editor), but for me it looks like the `worldLines.push(worldLine)`, etc.. should be out of loop, you are collecting lines for each two point..., but probably it is intended for `individualLineSegments` wright?

Reply:
Yes, this is only for individual lines, where every new point in the input generates a new segment (2 points) in the output.